app/services/order_cycles/distributed_products_service.rb

```diff
@@ -4,50 +4,19 @@
 # The stock-checking includes on_demand and stock level overrides from variant_overrides.
 
 module OrderCycles
-  class DistributedProductsService
+  class DistributedProductsService # rubocop:disable Metrics/ClassLength
     def initialize(distributor, order_cycle, customer)
       @distributor = distributor
       @order_cycle = order_cycle
       @customer = customer
     end
 
-    # TODO refactor products_taxons_relation and products_supplier_relation
-    # Joins on the first product variant to allow us to filter product by taxon.  # This is so
-    # enterprise can display product sorted by category in a custom order on their shopfront.
-    #
-    # Caveat, the category sorting won't work properly if there are multiple variant with different
-    # category for a given product.
-    #
-    def products_taxons_relation
-      Spree::Product.where(id: stocked_products).
-        joins("LEFT JOIN (
-                 SELECT DISTINCT ON(product_id) id, product_id, primary_taxon_id
-                 FROM spree_variants WHERE deleted_at IS NULL
-               ) first_variant ON spree_products.id = first_variant.product_id").
-        select("spree_products.*, first_variant.primary_taxon_id").
-        group("spree_products.id, first_variant.primary_taxon_id")
-    end
-
-    # Joins on the first product variant to allow us to filter product by supplier. This is so
-    # enterprise can display product sorted by supplier in a custom order on their shopfront.
-    #
-    # Caveat, the supplier sorting won't work properly if there are multiple variant with different
-    # supplier for a given product.
-    #
-    def products_supplier_relation
-      Spree::Product.where(id: stocked_products).
-        joins("LEFT JOIN (SELECT DISTINCT ON(product_id) id, product_id, supplier_id
-                          FROM spree_variants WHERE deleted_at IS NULL) first_variant
-                          ON spree_products.id = first_variant.product_id").
-        select("spree_products.*, first_variant.supplier_id").
-        group("spree_products.id, first_variant.supplier_id")
-    end
-
-    def supplier_property_join(query)
-      query.joins("
-        JOIN enterprises ON enterprises.id = first_variant.supplier_id
-        LEFT OUTER JOIN producer_properties ON producer_properties.producer_id = enterprises.id
-      ")
+    def products_relation(supplier_properties: nil)
```

Comment:
I could be wrong, but inside `DistributedProductsService` this seems to only be used for a switch (include properties and force sorting), so I think it could be a boolean value.